resources/audioPlayer.go

```diff
@@ -0,0 +1,106 @@
+package resources
+
+import (
+	"log"
+	"os"
+	"time"
+
+	"github.com/faiface/beep"
+	"github.com/faiface/beep/mp3"
+	"github.com/faiface/beep/speaker"
+)
+
+type sound struct {
+	index             int
+	path              string
+	isBackgroundMusic bool
+}
+
+// GetSounds gets all sounds for game
+func GetSounds() map[string]*sound {
+	return map[string]*sound{
+		"menu": {
+			index:             0,
+			path:              menuMusicPath,
+			isBackgroundMusic: true,
+		},
+		"game": {
+			index:             1,
+			path:              gameMusicPath,
+			isBackgroundMusic: true,
+		},
+		"laser": {
+			index:             2,
+			path:              laserSoundPath,
+			isBackgroundMusic: false,
+		},
+	}
+}
+
+// GetSoundFile returns streamer and format of decoded mp3 file
+func GetSoundFile(soundPath string) (beep.StreamSeekCloser, beep.Format) {
+	f, err := os.Open(soundPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	streamer, format, err := mp3.Decode(f)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return streamer, format
+}
+
+// GetSoundEffectsBuffer returns buffer of all game sound effects to avoid delays/load times
+// (cannot do same for background music as it consumes too much memory)
+func GetSoundEffectsBuffer() *beep.Buffer {
+	// declare buffer
+	var buffer *beep.Buffer
+
+	// loop over and add sound effects to buffer
+	sounds := GetSounds()
+	for k := range sounds {
+		thisSound := sounds[k]
+		if !thisSound.isBackgroundMusic {
+			streamer, format := GetSoundFile(thisSound.path)
+			if buffer == nil {
```

Comment:
wont this always be null?

Reply:
currently cuz there's only one sound effect, but in theory if we have a bunch then we should only declare the buffer once

Reply:
oh i see. nvm